robot/impl/local_robot.go

```diff
@@ -193,23 +193,23 @@ func (r *localRobot) StopAll(ctx context.Context, extra map[resource.Name]map[st
 	}
 
 	// Stop all stoppable resources
-	resourceErrs := []string{}
+	resourceErrs := make(map[string]error)
 	for _, name := range r.ResourceNames() {
 		res, err := r.ResourceByName(name)
 		if err != nil {
-			resourceErrs = append(resourceErrs, name.Name)
+			resourceErrs[name.Name] = err
 			continue
 		}
 
 		if actuator, ok := res.(resource.Actuator); ok {
 			if err := actuator.Stop(ctx, extra[name]); err != nil {
-				resourceErrs = append(resourceErrs, name.Name)
+				resourceErrs[name.Name] = err
 			}
 		}
 	}
 
-	if len(resourceErrs) > 0 {
-		return errors.Errorf("failed to stop components named %s", strings.Join(resourceErrs, ","))
+	for k, v := range resourceErrs {
+		return errors.Errorf("failed to stop component named %s with error %v", k, v)
```

Comment:
we should join all the errors first before returning them, right now you're returning at the first iteration of this for loop, and so only returning the first error. Not the same behaviour as before.

Reply:
oh meant to log this every time not return. is there a benefit to returning a joined error rather than just logging each error as we iterate through `resourceErrs`

Reply:
Since the behaviour before was to return an error, that would be a change in the behaviour of StopAll; now - if Stopall failed to stop your actuating components, you've got bigger problems than what the local robot does after. Session manager and App call stopAll for connectivity loss and moving away from the control page. If there are error checks there where they end up retrying if an error is encountered, then you're changing that behaviour of that caller.

Reply:
okay I'll wrap it all into one error